datumaro/components/dataset.py

```diff
@@ -807,6 +813,16 @@ def __init__(
         self._source_path = None
         self._options = {}
 
+    def __repr__(self) -> str:
+        return (
+            f"Dataset(size={len(self._data)}, "
+            f"source_path={self._source_path}, "
+            f"media_type={self.media_type()}, "
+            f"annotated_count={self.get_annotated_size()}, "
+            f"subset_count={len(self.subsets())}, "
+            f"categories_count={len(self.categories())})"
```

Comment:
I think this would be better to be more descriptive than now.
For example,
```
subsets
     train: # of items = x, # of annotations = y, annotation types = [Polygon, ...]
     val: # of items = z, # of annotations = w, annotation types = [Polygon, ...]
     ...:
}
categories = [0: "label1", 1: "label2", ....]
```

Reply:
Changed it like https://github.com/openvinotoolkit/datumaro/pull/750#issuecomment-1291551781.